source/views/news/news-list.tsx

```diff
@@ -50,11 +52,73 @@ export const NewsList = (props: Props): JSX.Element => {
 		isLoading,
 	} = useNews(props.source)
 
-	// remove all entries with blank excerpts
-	// remove all entries with a <form from the list
-	let entries = data
-		.filter((entry) => entry.excerpt.trim() !== '')
-		.filter((entry) => !entry.content.includes('<form'))
+	let entries = cleanEntries(data)
+
+	let [filters, setFilters] = React.useState<FilterType[]>([])
+	let getStoryCategories = React.useCallback((story: StoryType) => {
+		return story.categories.map((c) => trimStoryCateogry(c))
+	}, [])
+
+	let buildFilters = React.useCallback(() => {
+		let allCategories = entries.flatMap((story) => getStoryCategories(story))
+
+		if (allCategories.length === 0) {
+			setFilters([])
+			return
+		}
+
+		let categories = [...new Set(allCategories)].sort()
+		let filterCategories = categories.map((c) => {
+			return {title: c}
+		})
+
+		let newsFilters: ListType[] = [
+			{
+				type: 'list',
+				key: 'category', // TODO: figure if this is referencing anything or is simply unique
+				enabled: true,
+				spec: {
+					title: 'Categories',
+					options: filterCategories,
+					selected: filterCategories,
+					mode: 'OR',
+					displayTitle: true,
+				},
+				apply: {key: 'category'}, // TODO: figure out what this should reference
+			},
+		]
+		setFilters(newsFilters)
+
+		// Note:
+		// We have infinite re-rendering issues if we add 'entries' and 'getStoryCategories'
+		// to the dependency array. To reliably get this to re-render we need something to retrigger
+		// this callback.
+		//
+		// I recognize that:
+		// 1. disabling the linter for hooks is not great,
+		// 2. excluding two hook dependencies is less than ideal,
+		// 3. including an extraneous dependency is bad practice.
+		//
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [isPending])
```

Comment:
This may need some rethinking.

Reply:
Resolved.

Reply:
Do we see this as a blocker? Do we see this as able to merge and be iterated upon?